code/modules/projectiles/guns/flamer.dm

```diff
@@ -544,19 +544,36 @@
 	var/obj/screen/ammo/A = user.hud_used.ammo
 	A.update_hud(user)
 
-/obj/item/weapon/gun/flamer/marinestandard/Fire(atom/target, mob/living/user, params, reflex)
+/obj/item/weapon/gun/flamer/marinestandard/Fire()
 	if(active_attachable && istype(active_attachable, /obj/item/attachable/hydro_cannon) && (world.time > last_use + 10))
-		extinguish(target,user) //Fire it.
+		INVOKE_ASYNC(src, .proc/flamer_extinguish, target, gun_user) //Fire it.
 		water_count -=7//reagents is not updated in this proc, we need water_count for a updated HUD
 		last_fired = world.time
 		last_use = world.time
-		var/obj/screen/ammo/A = user.hud_used.ammo
-		A.update_hud(user)
-		return
-	if(user.skills.getRating("firearms") < 0 && !do_after(user, 1 SECONDS, TRUE, src))
+		var/obj/screen/ammo/A = gun_user.hud_used.ammo
+		A.update_hud(gun_user)
 		return
+	if(gun_user.skills.getRating("firearms"))
+		switch(windup_checked)
+			if(WEAPON_WINDUP_NOT_CHECKED)
+				INVOKE_ASYNC(src, .proc/do_windup)
+				return
+			if(WEAPON_WINDUP_CHECKING)
+				return
 	return ..()
 
+///Flamer windup called before firing
+/obj/item/weapon/gun/flamer/marinestandard/proc/do_windup()
+	windup_checked = WEAPON_WINDUP_CHECKING
+	if(!do_after(gun_user, 1 SECONDS, TRUE, src))
+		windup_checked = WEAPON_WINDUP_NOT_CHECKED
+		return
+	windup_checked = WEAPON_WINDUP_CHECKED
+	Fire()
```

Comment:
This is not skill-based anymore

Reply:
![image](https://user-images.githubusercontent.com/25491240/110256182-c9f50780-7f97-11eb-80a9-b29d4dc6e899.png)

It is